src/docfx/build/document/DocumentProvider.cs

```diff
@@ -256,20 +256,22 @@ private static string PathToRelativeUrl(string path, ContentType contentType, Ur
             }
             if (urlType == UrlType.Docs && contentType != ContentType.Toc)
             {
+                // remove extension
                 var i = url.LastIndexOf('.');
                 return i >= 0 ? url[..i] : url;
             }
         }
         return url;
     }
 
-    /// <summary>
-    /// In docs, canonical URL is later overwritten by template JINT code.
-    /// TODO: need to handle the logic difference when template code is removed.
-    /// </summary>
-    private string GetCanonicalUrl(string siteUrl)
+    private string GetCanonicalUrl(string? siteUrl)
     {
-        return $"https://{_config.HostName}/{_buildOptions.Locale}{siteUrl}";
+        if (siteUrl == null)
+        {
+            return "";
+        }
+
+        return $"https://{_config.HostName}/{_buildOptions.Locale}{UrlUtility.EscapeUrlPath(siteUrl).ToLowerInvariant()}";
```

Comment:
Do we want to lower case the query string?

Reply:
at this moment, the return value still has no query value (i.e. moniker) 
The moniker query is added on PageBuilder